### ISSUE TITLE
Prevent editor from stealing focus when link is clicked in terminal

### ISSUE THREAD

**Issue by Moshyfawn** [zed-industries/zed#9726]
### Check for existing issues

- [X] Completed

### Describe the feature

I tried to navigate to the terminal by right clicking on it to focus. I had several file paths displayed due to a previous drag. I accidentally clicked on one of the many links that caused the editor to steal focus, which I didn't notice and started typing my command.

I understand the convenience of having links open files directly in the editor, but perhaps the action should be "hidden" behind a `cmd-click` key combination instead, because in case most of the terminal is filled with links, it'd be tedious to find a place to focus the editor with the mouse.

### If applicable, add mockups / screenshots to help present your vision of the feature

https://github.com/zed-industries/zed/assets/16290753/dff770cb-bc32-4241-a8ef-f806049ec450

**Comment by danielzsh** [comment]
I ran into this just now, and it seems like a bug; before doing cmd + click once, a normal left-click doesn't open the file/folder,  but after one cmd + click, normal clicks begin to be treated as "cmd + click" as well, with a notable lack of the underline that appears when hovering with "cmd" held down.

https://github.com/zed-industries/zed/assets/65467530/34f78c56-95be-4566-9feb-127f563f3a01

**Comment by mayfieldiv** [comment]
It looks like the problem is that when you cmd-click, the terminal loses focus and isn't notified when you release the cmd modifier (while focused on e.g. the editor), so the terminal behaves as if cmd is still pressed on subsequent clicks until you press and release cmd inside the terminal

**Comment by danielzsh** [comment]
Good catch! I think if we implement some kind of focus follows mouse as per #8167, that should resolve this issue as well (basically as long as we have our mouse over the terminal, it'll still have focus and thus be able to listen to modifiers) but I'm not quite sure how to implement that yet...

**Comment by notpeter** [comment]
I believe this has been fixed.
